src/main/scala/xerial/sbt/Sonatype.scala

```diff
@@ -31,6 +31,14 @@ object Sonatype extends AutoPlugin with LogSupport {
     val sonatypePublishTo               = settingKey[Option[Resolver]]("Default Sonatype publishTo target")
     val sonatypePublishToBundle         = settingKey[Option[Resolver]]("Default Sonatype publishTo target")
     val sonatypeTargetRepositoryProfile = settingKey[StagingRepositoryProfile]("Stating repository profile")
+    val sonatypeTopLevelGroupIdAsProfileName =
+      settingKey[Boolean](
+        """Whether to use only top-level groupId as the staging profile name (e.g. only `com.company`, even though the whole groupId is `com.company.project.product`).
+          |Note that groupIds like `com.github.*` or `uk.*` are exception and the staging profile name will contain 3 parts, so `io.gitlab.team.project` will result to profile `io.gitlab.team`.
+          |This is suitable because Sonatype prefers to create only one staging profile, which corresponds to the top-level groupId and then every other artifact which has this groupId prefix can be published there.
```

Comment:
@jorlina can you please confirm that this is indeed the case?

Reply:
I want to point out that preparing staging repositories for top-level groupIds is not a hard requirement from the Sonatype end.  Introducing potentially breaking changes to actual tooling to support this preference doesn't seem right to me.  I think the real solution is to better educate users of OSSRH when they're signing up for groupIds that they only need to request the top-level groupId, which, as you're example show aren't restricted to xxx.yyy.

Reply:
Please note, that this feature is not on by default :wink: